pyanaconda/modules/storage/devicetree/viewer.py

```diff
@@ -347,13 +347,16 @@ def resolve_device(self, dev_spec):
         The spec can be anything from a device name (eg: 'sda3') to a
         device node path (eg: '/dev/mapper/fedora-root') to something
         like 'UUID=xyz-tuv-qrs' or 'LABEL=rootfs'.
+        For btrfs subvolumes the subvolume can be specified by the UUID of
+        the volume and the name of the subvolume, as:
+        'UUID=2252ec30-1fce-4f8e-bdef-c50c3a44ede4@root'
 
         If no device is found, return an empty string.
 
         :param dev_spec: a string describing a block device
         :return: a device name or an empty string
         """
-        device = self.storage.devicetree.resolve_device(dev_spec)
+        device = resolve_device(self.storage, dev_spec)
```

Comment:
It kind of scares me to have two different versions of `resolve_device`. I would be fine with overriding the `DeviceTree` class with our own and using the new version everywhere (if it is safe to do so).